src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -243,6 +253,15 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             if (mode == Mode.GlobbedHref || mode == Mode.Fallback && !string.IsNullOrEmpty(HrefInclude))
             {
+                if (TryResolveUrl(HrefInclude, tryEncodeApplicationPath: false, resolvedUrl: out resolvedUrl))
```

Comment:
We don't support ~/ today in globbed patterns. https://github.com/aspnet/Mvc/issues/2903